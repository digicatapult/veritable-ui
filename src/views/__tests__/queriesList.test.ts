import { expect } from 'chai'
import { describe, it } from 'mocha'
import QueryListTemplates from '../queriesList.js'

describe('ConnectionTemplates', () => {
  const sampleDate = new Date()

  describe('listPage', () => {
    it('should render with no connections', async () => {
      const templates = new QueryListTemplates()
      const rendered = await templates.listPage([])
      expect(rendered).to.matchSnapshot()
    })
    it('should render with single query', async () => {
      const templates = new QueryListTemplates()
      const rendered = await templates.listPage([
        {
          company_name: 'Company A',
          status: 'resolved',
          query_type: 'Type A',
          updated_at: sampleDate,
        },
      ])
      expect(rendered).to.matchSnapshot()
    })
    it('should escape html in name', async () => {
      const templates = new QueryListTemplates()
      const rendered = await templates.listPage([
        {
          company_name: '<div>I own you</div>',
          status: 'resolved',
          query_type: 'Type A',
          updated_at: sampleDate,
        },
      ])
      expect(rendered).to.matchSnapshot()
    })
    it('should render multiple with each status', async () => {
      const templates = new QueryListTemplates()
      const rendered = await templates.listPage([
        {
          company_name: 'Company A',
          status: 'resolved',
          query_type: 'Type A',
          updated_at: sampleDate,
        },
        {
          company_name: 'Company B',
          status: 'resolved',
          query_type: 'Type A',
          updated_at: sampleDate,
        },
        {
          company_name: 'Company C',
          status: 'resolved',
          query_type: 'Type A',
          updated_at: sampleDate,
        },
        {
          company_name: 'Company D',
          status: 'resolved',
          query_type: 'Type A',
          updated_at: sampleDate,
        },
        {
          company_name: 'Company E',
          status: 'resolved',
          query_type: 'Type A',
          updated_at: sampleDate,
        },
        {
          company_name: 'Company F',
          status: 'resolved',
          query_type: 'Type A',
          updated_at: sampleDate,
        },
      ])
      expect(rendered).to.matchSnapshot()
    })
  })
})
