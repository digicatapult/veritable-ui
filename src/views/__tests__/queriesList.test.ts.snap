// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionTemplates listPage should escape html in name 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx.org/ext/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><title>Veritable - Queries</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><nav class=\\"flex-page side-bar\\"><img class=\\"side-bar logo-container\\" src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" class=\\"side-bar icon\\" style=\\"background-image:url(&#34;/public/images/category.svg&#34;);\\"></a><a title=\\"connections\\" href=\\"/connection\\" class=\\"side-bar icon active\\" style=\\"background-image:url(&#34;/public/images/connection.svg&#34;);\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/folder.svg&#34;);\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/notification.svg&#34;);\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/setting.svg&#34;);\\"></a></nav><main class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\">Query Management</h1><div class=\\"header nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><span class=\\"url-separator\\">/</span><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a></div></div><div class=\\"content main\\"><div class=\\"main-list-page\\" hx-get=\\"/queries\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#queries-search-input\\"><div class=\\"list-page-header\\"><span>Query Management</span><a class=\\"button   button-filled\\" href=\\"/queries/new\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/plus.svg&#34;);\\"></div><span class=\\"button-text \\">Query Request</span></a></div><div class=\\"list-page \\"><div class=\\"list-nav\\"><span>Query Management</span><input id=\\"queries-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search\\" hx-get=\\"/queries\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"list-page\\"><thead><th><span>Company Name</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Query Type</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Direction</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Requested deadline</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Verification Status</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Actions</span><a class=\\"list-table icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>&lt;div>I own you&lt;/div></td><td>Type A</td><td><p>Received</p></td><td><time datetime=\\"2024-07-04T00:00:00.000Z\\">7/4/2024, 2:00:00 AM</time></td><td><div class=\\"success\\">Resolved</div></td><td><a class=\\"button disabled outline button-filled\\" href=\\"#\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span class=\\"button-text accent\\">some action</span></a></td></tr></tbody></table></div></div></div></main></body></html>"`;

exports[`ConnectionTemplates listPage should render multiple with each status 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx.org/ext/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><title>Veritable - Queries</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><nav class=\\"flex-page side-bar\\"><img class=\\"side-bar logo-container\\" src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" class=\\"side-bar icon\\" style=\\"background-image:url(&#34;/public/images/category.svg&#34;);\\"></a><a title=\\"connections\\" href=\\"/connection\\" class=\\"side-bar icon active\\" style=\\"background-image:url(&#34;/public/images/connection.svg&#34;);\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/folder.svg&#34;);\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/notification.svg&#34;);\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/setting.svg&#34;);\\"></a></nav><main class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\">Query Management</h1><div class=\\"header nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><span class=\\"url-separator\\">/</span><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a></div></div><div class=\\"content main\\"><div class=\\"main-list-page\\" hx-get=\\"/queries\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#queries-search-input\\"><div class=\\"list-page-header\\"><span>Query Management</span><a class=\\"button   button-filled\\" href=\\"/queries/new\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/plus.svg&#34;);\\"></div><span class=\\"button-text \\">Query Request</span></a></div><div class=\\"list-page \\"><div class=\\"list-nav\\"><span>Query Management</span><input id=\\"queries-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search\\" hx-get=\\"/queries\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"list-page\\"><thead><th><span>Company Name</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Query Type</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Direction</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Requested deadline</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Verification Status</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Actions</span><a class=\\"list-table icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>Company A</td><td>Type A</td><td><p>Received</p></td><td><time datetime=\\"2024-07-04T00:00:00.000Z\\">7/4/2024, 2:00:00 AM</time></td><td><div class=\\"success\\">Resolved</div></td><td><a class=\\"button disabled outline button-filled\\" href=\\"#\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span class=\\"button-text accent\\">some action</span></a></td></tr><tr><td>Company B</td><td>Type A</td><td><p>Received</p></td><td><time datetime=\\"2024-07-04T00:00:00.000Z\\">7/4/2024, 2:00:00 AM</time></td><td><div class=\\"success\\">Resolved</div></td><td><a class=\\"button disabled outline button-filled\\" href=\\"#\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span class=\\"button-text accent\\">some action</span></a></td></tr><tr><td>Company C</td><td>Type A</td><td><p>Received</p></td><td><time datetime=\\"2024-07-04T00:00:00.000Z\\">7/4/2024, 2:00:00 AM</time></td><td><div class=\\"success\\">Resolved</div></td><td><a class=\\"button disabled outline button-filled\\" href=\\"#\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span class=\\"button-text accent\\">some action</span></a></td></tr><tr><td>Company D</td><td>Type A</td><td><p>Received</p></td><td><time datetime=\\"2024-07-04T00:00:00.000Z\\">7/4/2024, 2:00:00 AM</time></td><td><div class=\\"success\\">Resolved</div></td><td><a class=\\"button disabled outline button-filled\\" href=\\"#\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span class=\\"button-text accent\\">some action</span></a></td></tr><tr><td>Company E</td><td>Type A</td><td><p>Received</p></td><td><time datetime=\\"2024-07-04T00:00:00.000Z\\">7/4/2024, 2:00:00 AM</time></td><td><div class=\\"success\\">Resolved</div></td><td><a class=\\"button disabled outline button-filled\\" href=\\"#\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span class=\\"button-text accent\\">some action</span></a></td></tr><tr><td>Company F</td><td>Type A</td><td><p>Received</p></td><td><time datetime=\\"2024-07-04T00:00:00.000Z\\">7/4/2024, 2:00:00 AM</time></td><td><div class=\\"success\\">Resolved</div></td><td><a class=\\"button disabled outline button-filled\\" href=\\"#\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span class=\\"button-text accent\\">some action</span></a></td></tr></tbody></table></div></div></div></main></body></html>"`;

exports[`ConnectionTemplates listPage should render with no connections 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx.org/ext/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><title>Veritable - Queries</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><nav class=\\"flex-page side-bar\\"><img class=\\"side-bar logo-container\\" src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" class=\\"side-bar icon\\" style=\\"background-image:url(&#34;/public/images/category.svg&#34;);\\"></a><a title=\\"connections\\" href=\\"/connection\\" class=\\"side-bar icon active\\" style=\\"background-image:url(&#34;/public/images/connection.svg&#34;);\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/folder.svg&#34;);\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/notification.svg&#34;);\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/setting.svg&#34;);\\"></a></nav><main class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\">Query Management</h1><div class=\\"header nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><span class=\\"url-separator\\">/</span><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a></div></div><div class=\\"content main\\"><div class=\\"main-list-page\\" hx-get=\\"/queries\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#queries-search-input\\"><div class=\\"list-page-header\\"><span>Query Management</span><a class=\\"button   button-filled\\" href=\\"/queries/new\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/plus.svg&#34;);\\"></div><span class=\\"button-text \\">Query Request</span></a></div><div class=\\"list-page \\"><div class=\\"list-nav\\"><span>Query Management</span><input id=\\"queries-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search\\" hx-get=\\"/queries\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"list-page\\"><thead><th><span>Company Name</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Query Type</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Direction</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Requested deadline</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Verification Status</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Actions</span><a class=\\"list-table icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>No Queries for that search. Try again or add a new query</td></tr></tbody></table></div></div></div></main></body></html>"`;

exports[`ConnectionTemplates listPage should render with single query 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx.org/ext/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><title>Veritable - Queries</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><nav class=\\"flex-page side-bar\\"><img class=\\"side-bar logo-container\\" src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" class=\\"side-bar icon\\" style=\\"background-image:url(&#34;/public/images/category.svg&#34;);\\"></a><a title=\\"connections\\" href=\\"/connection\\" class=\\"side-bar icon active\\" style=\\"background-image:url(&#34;/public/images/connection.svg&#34;);\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/folder.svg&#34;);\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/notification.svg&#34;);\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"side-bar icon disabled\\" style=\\"background-image:url(&#34;/public/images/setting.svg&#34;);\\"></a></nav><main class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\">Query Management</h1><div class=\\"header nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><span class=\\"url-separator\\">/</span><a title=\\"Query Management\\" href=\\"/queries\\">Query Management</a></div></div><div class=\\"content main\\"><div class=\\"main-list-page\\" hx-get=\\"/queries\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#queries-search-input\\"><div class=\\"list-page-header\\"><span>Query Management</span><a class=\\"button   button-filled\\" href=\\"/queries/new\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/plus.svg&#34;);\\"></div><span class=\\"button-text \\">Query Request</span></a></div><div class=\\"list-page \\"><div class=\\"list-nav\\"><span>Query Management</span><input id=\\"queries-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search\\" hx-get=\\"/queries\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"list-page\\"><thead><th><span>Company Name</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Query Type</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Direction</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Requested deadline</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Verification Status</span><a class=\\"list-table icon disabled\\"></a></th><th><span>Actions</span><a class=\\"list-table icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>Company A</td><td>Type A</td><td><p>Received</p></td><td><time datetime=\\"2024-07-04T00:00:00.000Z\\">7/4/2024, 2:00:00 AM</time></td><td><div class=\\"success\\">Resolved</div></td><td><a class=\\"button disabled outline button-filled\\" href=\\"#\\"><div class=\\"button-icon\\" style=\\"background-image:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span class=\\"button-text accent\\">some action</span></a></td></tr></tbody></table></div></div></div></main></body></html>"`;
